DockService.Infrastructure/Services/DockService.cs

```diff
@@ -0,0 +1,61 @@
+﻿using DockService.Core.Messaging;
+using DockService.Core.Models;
+using DockService.Core.Repositories;
+using DockService.Core.Services;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DockService.Infrastructure.Services
+{
+    public class DockService : IDockService
+    {
+        private readonly IShipRepository _shipRepository;
+        private readonly IEventPublisher _eventPublisher;
+
+        public DockService(IShipRepository shipRepository, IEventPublisher eventPublisher)
+        {
+            _shipRepository = shipRepository;
+            _eventPublisher = eventPublisher;
+        }
+
+        #region -- DB operations --
+        public Task<Ship> CreateShipAsync(Ship ship)
+        {
+            return _shipRepository.CreateShip(ship);
+        }
+
+        public Task DeleteShipAsync(Guid shipId)
+        {
+            return _shipRepository.DeleteShip(shipId);
+        }
+
+        public Task<Ship> GetShipAsync(Guid shipId)
+        {
+            return _shipRepository.GetShip(shipId);
+        }
+        #endregion
+
+        #region -- Events --
+        public async Task SendShipDockedAsync(Ship ship)
+        {
+            Console.WriteLine("Docking ship: " + ship.Id);
+            await _eventPublisher.HandleEventAsync(EventTypes.ShipDocked, ship);
```

Comment:
Send ShipId and CustomerId in 1 object

Reply:
i dont have customer data whatsoever

Reply:
added customerId to ship anyway, it will be passed along